biz.aQute.bndlib/src/aQute/bnd/header/Attrs.java

```diff
@@ -214,7 +214,13 @@ public void putTyped(String key, Object value) {
 			} else {
 				type = getObjectType(value);
 			}
-			key += ":" + type.toString();
+			if (key.endsWith(":")) {
+				// This is a directive
+				key += type.toString() + ":";
```

Comment:
This is wrong. Directives are never typed. They are always String type. Only attributes can be typed. See https://osgi.org/specification/osgi.core/7.0.0/framework.module.html#d0e2821.

Reply:
That's good to know - in which case the Attrs is still doing the wrong thing. Should it be an exception to putTyped with a Directive (ending in a `:`), or should it simply not add a type?

Reply:
For reference, the existing behaviour is to output

    x-directiveName::List<String>="foo, bar"

when it encounters something like:

    @Directive("x-directiveName")
    String[] value;

Reply:
I am working on a change to Attrs now...

Reply:
My change is in https://github.com/bndtools/bnd/pull/2435

Reply:
@timothyjward I merged #2435 so you can rebase on the tip of master removing your changes to Attrs. My change just avoids writing the type for directives since they must be String. So 

```
@Directive("x-directiveName")
String[] value;
```

should be emitted as:

```
x-directiveName:="foo,bar"
```

But frankly using a type other than String for an @Directive annotation element is strange.